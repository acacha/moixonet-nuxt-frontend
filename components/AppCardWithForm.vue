<template>
  <v-card class="elevation-12">
    <v-toolbar
      color="primary"
      dark
      flat
    >
      <v-toolbar-title data-test="card_title">
        <slot name="title">
          PUT YOUR TITLE HERE
        </slot>
      </v-toolbar-title>
    </v-toolbar>
    <span data-test="card_form">
      <slot name="form" />
    </span>
  </v-card>
</template>

<script>
export default {
  name: 'AppCardWithForm'
}
</script>
