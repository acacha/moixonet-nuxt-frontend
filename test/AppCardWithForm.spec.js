import { mount } from '@vue/test-utils'
import Vuetify from 'vuetify'
import Vue from 'vue'
import AppCardWithForm from '../components/AppCardWithForm.vue'

Vue.use(Vuetify)
describe('AppCardWithForm', () => {
  let opts
  const build = (options) => {
    options = options || opts
    const wrapper = mount(AppCardWithForm, options)
    return {
      wrapper,
      cardTitle: () => wrapper.find('[data-test=card_title]'),
      cardForm: () => wrapper.find('[data-test=card_form]')
    }
  }

  beforeEach(() => {
    opts = {
      slots: {
        title: '<div class="fake-title">TITLE</div>',
        form: '<div class="fake-form">TITLE</div>'
      },
      vuetify: new Vuetify({
        mocks: {
          $vuetify: {
            breakpoint: {} // STUB BREAKPOINTS SERVICE: https://vuetifyjs.com/en/getting-started/unit-testing#mocking-vuetify
          }
        }
      })
    }
  })

  test('is a Vue instance', () => {
    const { wrapper } = build()
    expect(wrapper.isVueInstance()).toBeTruthy()
  })

  test('renders the component with default slots', () => {
    const { wrapper } = build()
    expect(wrapper.element).toMatchSnapshot()
  })

  test('renders the component with slots', () => {
    const { wrapper } = build()
    expect(wrapper.element).toMatchSnapshot()
  })

  it('Title are inserted in v-toolbar-title element', () => {
    const { cardTitle } = build()
    expect(cardTitle().findAll('.fake-title').length).toBe(1)
  })

  it('Form are inserted inside card element', () => {
    const { cardForm } = build()
    expect(cardForm().findAll('.fake-form').length).toBe(1)
  })
})
