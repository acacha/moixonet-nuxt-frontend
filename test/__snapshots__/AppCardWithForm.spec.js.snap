// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`AppCardWithForm renders the component with default slots 1`] = `
<div
  class="elevation-12 v-card v-sheet theme--light"
>
  <header
    class="v-sheet v-sheet--tile theme--dark v-toolbar v-toolbar--flat primary"
    style="height: 64px;"
  >
    <div
      class="v-toolbar__content"
      style="height: 64px;"
    >
      <div
        class="v-toolbar__title"
        data-test="card_title"
      >
        <div
          class="fake-title"
        >
          TITLE
        </div>
      </div>
    </div>
  </header>
   
  <span
    data-test="card_form"
  >
    <div
      class="fake-form"
    >
      TITLE
    </div>
  </span>
</div>
`;

exports[`AppCardWithForm renders the component with slots 1`] = `
<div
  class="elevation-12 v-card v-sheet theme--light"
>
  <header
    class="v-sheet v-sheet--tile theme--dark v-toolbar v-toolbar--flat primary"
    style="height: 64px;"
  >
    <div
      class="v-toolbar__content"
      style="height: 64px;"
    >
      <div
        class="v-toolbar__title"
        data-test="card_title"
      >
        <div
          class="fake-title"
        >
          TITLE
        </div>
      </div>
    </div>
  </header>
   
  <span
    data-test="card_form"
  >
    <div
      class="fake-form"
    >
      TITLE
    </div>
  </span>
</div>
`;
